components/CustomGoogleLoginButton.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { useGoogleLogin } from '@react-oauth/google';
+import { axiosInstance } from '@/utils/axios';
+import useAuthResonse from '@/hooks/useAuthResponse';
+
+interface CustomGoogleLoginButtonProps {
+    updateLoading: (isLoading: boolean) => void;
+  }
+
+const CustomGoogleLoginButton: React.FC<CustomGoogleLoginButtonProps> = ({ updateLoading }) => {
+
+    const handleAuthResponse = useAuthResonse();
+
+    const googleLogin = useGoogleLogin({
+        onSuccess: codeResponse => exchangeCodeForTokens(codeResponse.code),
+        onError: () => { console.log('Login failed'); updateLoading(false); }, // TODO implement error handling
+        flow: "auth-code",
+    });
+
+  const exchangeCodeForTokens = async (code: string) => {
+    try {
+        const response = await axiosInstance.get<any>(
+            `/auth/google/grant-code?code=${code}`
+          );
+          if (!response) {
+            throw new Error('Failed to exchange code for tokens');
+          }
+          handleAuthResponse(response.data.token, response.data.refreshToken);
+      
+          // Handle the response, e.g., store the tokens, update UI, etc.
+          console.log('Tokens:', response.data);
+    } catch (error) {
+        console.error(error);
+    } finally {
+        updateLoading(false);
+    }
+  };
+
+  return (
+      <button className="gsi-material-button" onClick={() => {
+          updateLoading(true);
+          googleLogin();
+      }}>
+  <div className="gsi-material-button-state"></div>
```

Comment:
The Style below is generated by google
https://developers.google.com/identity/branding-guidelines